src/openassetio-core/src/hostApi/Manager.cpp

```diff
@@ -39,7 +39,8 @@ void throwFromBatchElementError(std::size_t index, BatchElementError error) {
       exceptionMessage += std::to_string(static_cast<int>(error.code));
       exceptionMessage += " Message: ";
       exceptionMessage += error.message;
-      throw errors::InputValidationException{exceptionMessage};
+      error.message = std::move(exceptionMessage);
+      throw UnknownBatchElementException{index, std::move(error)};
```

Comment:
[Consider] I guess this was the subject of discussions, so we don't need to re-legislate, but I'll register that this seems off to me, we already have an `unknown` code that converts to the unknown exception. ... sometimes i wonder if cases like this should just be `assert(false)`s to indicate programming error.

Reply:
True it is a programming error, `assert` might be good but obviously won't work in Release builds.

The change here was because the exception propagates to the host, when it was the manager that made the error. So from the host's perspective it really is an "unknown" error.  Should we have another exception type, just for this case, perhaps?  The message and index might still be useful to hosts.

Reply:
Na, a new exception type isn't worth it, this isn't even a real case, i'm convinced it can't happen with our clang checks enable, just leave it.